src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingSwitchDefaultCheck.java

```diff
@@ -170,7 +170,8 @@ private static boolean containsNullCaseLabelElement(DetailAST detailAst) {
      */
     private static boolean isSwitchExpression(DetailAST ast) {
         return ast.getParent().getType() == TokenTypes.EXPR
-                || ast.getParent().getParent().getType() == TokenTypes.EXPR;
+                || ast.getParent().getParent().getType() == TokenTypes.EXPR
+                || ast.getParent().getParent().getParent().getType() == TokenTypes.EXPR;
```

Comment:
This seems like we might be getting close to bad code. How do you know we aren't going to find an expression token and the switch isn't a switch expression?

This is the closest example I could think up, but the expression token is 4 parents up and not 3.
````
public class TestClass {
    void method() {
        method(() -> {
        switch (value) {  // violation, 'switch without "default" clause.'
            case ONE -> System.out.println("one");
            case TWO -> System.out.println("two");
        }
        });
    }
}
````

Is there no other way to verify we are dealing with a switch statement over a switch expression except to locate an expression token?

Reply:
I think it will be clearer and more robust if we can check it like this: 
`it is a switch expression if it is not a switch statement`.
 if the parent of the switch is a `SLIST` then it is a switch **statement**, switch statements are always standalone and thus direct children of a `SLIST` in the AST. but switch expressions are used within expressions like assignments or returns, their parent can't be `SLIST`. I think this can effectively distinguish between switch statements and switch expressions. let's see if regression report can confirm that